qemu/tests/thin_write_in_qemu_img_commit.py

```diff
@@ -0,0 +1,60 @@
+import logging
+import json
+
+from virttest import data_dir
+from virttest import utils_numeric
+from virttest.qemu_io import QemuIOSystem
+from virttest.qemu_storage import QemuImg
+
+from qemu.tests.qemu_disk_img import QemuImgTest
+from qemu.tests.qemu_disk_img import generate_base_snapshot_pair
+
+
+def run(test, params, env):
+    """
+    Check thin write in qemu-img commit.
+
+    :param test: Qemu test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    def _get_image_object(tag):
+        """Get QemuImg object by tag."""
+        img_param = params.object_params(tag)
+        return QemuImg(img_param, img_root_dir, tag)
+
+    def _create_external_snapshot(tag):
+        """Create an external snapshot based on tag."""
+        logging.info("Create external snapshot %s." % tag)
+        qit = QemuImgTest(test, params, env, snapshot)
+        qit.create_snapshot()
+
+    def _qemu_io(img, cmd):
+        """Run qemu-io cmd to a given img."""
+        logging.info("Run qemu-io %s" % img.image_filename)
+        q = QemuIOSystem(test, params, img.image_filename)
+        q.cmd_output(cmd, 120)
+
+    def _verify_map_output(output):
+        """"Verify qemu map output."""
+        expected = {
+            "length": int(utils_numeric.normalize_data_size(
+                params["write_size"], "B")),
+            "start": 0, "depth": 0, "zero": True, "data": False}
+        if expected not in json.loads(output.stdout_text):
+            test.fail("There should be no data from 0 to %s." %
+                      params["write_size"])
```

Comment:
It would be better "The map info is inaccurate from 0 to %s of the %s image file. " ?

Reply:
updated thanks